presto-native-execution/presto_cpp/main/TaskManager.cpp

```diff
@@ -333,11 +333,8 @@ std::unique_ptr<TaskInfo> TaskManager::createOrUpdateTask(
 
   getDataForResultRequests(resultRequests);
 
-  // TODO Handle possible race condition. Refer:
-  // https://github.com/facebookincubator/velox/issues/3593
-  if (outputBuffers.type == protocol::BufferType::BROADCAST) {
-    execTask->updateBroadcastOutputBuffers(
-        outputBuffers.buffers.size(), outputBuffers.noMoreBufferIds);
+  if (!updateBroadcastOutputBuffers(prestoTask, execTask, outputBuffers)) {
```

Comment:
Why are you always calling updateBroadcastOutputBuffers ? It seems more readable to check the BROADCAST condition here than in the function called ?